experimental/compactdecimal/src/compactdecimal.rs

```diff
@@ -0,0 +1,235 @@
+    ///
+    /// ```
+    /// use icu_compactdecimal::CompactDecimalFormatter;
+    /// use icu_decimal::options::FixedDecimalFormatterOptions;
+    /// use icu_locid::locale;
+    ///
+    /// CompactDecimalFormatter::try_new_long_unstable(
+    ///     &icu_testdata::unstable(),
+    ///     &locale!("sv").into(),
+    ///     FixedDecimalFormatterOptions::default());
+    /// ```
+    ///
+    /// [data provider]: icu_provider
+    pub fn try_new_long_unstable<D>(
+        data_provider: &D,
+        locale: &DataLocale,
+        options: FixedDecimalFormatterOptions,
+    ) -> Result<Self, CompactDecimalError>
+    where
+        D: DataProvider<LongCompactDecimalFormatDataV1Marker>
+            + DataProvider<icu_decimal::provider::DecimalSymbolsV1Marker>
+            + DataProvider<icu_plurals::provider::CardinalV1Marker>
+            + ?Sized,
+    {
+        Ok(Self {
+            fixed_decimal_format: FixedDecimalFormatter::try_new_unstable(
+                data_provider,
+                locale,
+                options,
+            )?,
+            plural_rules: PluralRules::try_new_cardinal_unstable(data_provider, locale)?,
+            compact_data: DataProvider::<LongCompactDecimalFormatDataV1Marker>::load(
+                data_provider,
+                DataRequest {
+                    locale,
+                    metadata: Default::default(),
+                },
+            )?
+            .take_payload()?
+            .cast(),
+        })
+    }
+
+    /// TODO(egg): Explain that this is an advanced API which can fail (and you
+    /// should use the API that does not exist instead).
+    ///
+    /// # Examples
+    /// ```
+    /// use fixed_decimal::CompactDecimal;
+    /// use icu_compactdecimal::CompactDecimalFormatter;
+    /// use icu_decimal::options::FixedDecimalFormatterOptions;
+    /// use icu_locid::locale;
+    /// use writeable::assert_writeable_eq;
+    /// use std::str::FromStr;
+    ///
+    /// let long_bangla = CompactDecimalFormatter::try_new_long_unstable(
+    ///     &icu_testdata::unstable(),
+    ///     &locale!("bn").into(),
+    ///     FixedDecimalFormatterOptions::default(),
+    /// ).unwrap();
+    ///
+    /// let short_french = CompactDecimalFormatter::try_new_short_unstable(
+    ///     &icu_testdata::unstable(),
+    ///     &locale!("fr").into(),
+    ///     FixedDecimalFormatterOptions::default(),
+    /// ).unwrap();
+    ///
+    /// let long_french = CompactDecimalFormatter::try_new_long_unstable(
+    ///     &icu_testdata::unstable(),
+    ///     &locale!("fr").into(),
+    ///     FixedDecimalFormatterOptions::default(),
+    /// ).unwrap();
+    ///
+    /// let about_a_million = CompactDecimal::from_str("1.2c6").unwrap();
+    /// let three_million = CompactDecimal::from_str("3c6").unwrap();
+    /// let ten_lakhs = CompactDecimal::from_str("10c5").unwrap();
+    ///
+    /// assert_writeable_eq!(short_french.format_compact_decimal(&about_a_million).unwrap(), "1,2\u{A0}M");
```

Comment:
Nit: In docstests only, we discussed last week whether to render whitespace and invisible characters; here is what we decided for docstests (https://github.com/unicode-org/icu4x/pull/2874#discussion_r1050009499):

```
    /// # // The following line contains a NBSP (U+A0)
    /// assert_writeable_eq!(short_french.format_compact_decimal(&about_a_million).unwrap(), "1,2 M");
```